### DISCUSSION TITLE
Authenticate requests with Web API Key (Client Mode)

### DISCUSSION THREAD

**Discussion by Mechstud** [kreait/firebase-php#422] [Ideas]
I am working on some client project. For Firebase, all they can give me is authorization key like the following:

```
key=AAAAAzz7EPs:APA91bafsafsfasfsafasfas_iAKWrsfIY09yD5NE_CpoLydasdaphEPct_bscw-Z70usadasdsqKiQNqtVcRysnasdasdasdasdasCbUudAzu8u
```

How can I create Factory object just using this key, and not the JSON key file ?

I am actually trying to modify the legacy cURL call to firebase (the abovementioned key is passed into AUTHORIZATION header). Hoping to be able to use this awesome library instead. Kindly help. 

Thanks in advance.

**Comment by jeromegamez** [comment]
The SDK is an _Admin_ SDK 😅, so using the web API key is currently not possible and I was opposed to support it for the longest time 🙈.

BUT: I have recently started working on making this possible. As I'm currently flooded with non-open-source work, I can't tell you when this will be available, but it will most likely not be before next month.

🙏
